src/Bridge/Doctrine/Orm/Util/QueryNameGenerator.php

```diff
@@ -11,175 +11,31 @@
 
 namespace ApiPlatform\Core\Bridge\Doctrine\Orm\Util;
 
-use Doctrine\Common\Persistence\ManagerRegistry;
-use Doctrine\ORM\Mapping\ClassMetadata;
-use Doctrine\ORM\Query\Expr\Join;
-use Doctrine\ORM\Query\Expr\OrderBy;
-use Doctrine\ORM\QueryBuilder;
-
 /**
  * Utility functions for working with Doctrine ORM query.
  *
  * @author Teoh Han Hui <teohhanhui@gmail.com>
  * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+
  */
-abstract class QueryNameGenerator
+final class QueryNameGenerator implements QueryNameGeneratorInterface
 {
-    /**
-     * Generates a unique alias for DQL join.
-     *
-     * @param string $association
-     *
-     * @return string
-     */
-    public static function generateJoinAlias(string $association) : string
-    {
-        return sprintf('%s_%s', $association, uniqid());
-    }
+    protected $incrementedQuery = 1;
 
     /**
-     * Generates a unique parameter name for DQL query.
-     *
-     * @param string $name
-     *
-     * @return string
+     * {@inheritdoc}
      */
-    public static function generateParameterName(string $name) : string
+    public function generateJoinAlias(string $association) : string
     {
-        return sprintf('%s_%s', $name, uniqid());
+        return sprintf('%s_%s', $association, $association.$this->incrementedQuery++);
```

Comment:
You should use 2 separate counter, one for associations and another one for parameters.